firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -119,11 +122,19 @@ public Thread newThread(Runnable r) {
       RandomFidGenerator fidGenerator) {
     this.firebaseApp = firebaseApp;
     this.serviceClient = serviceClient;
-    this.executor = executor;
     this.persistedInstallation = persistedInstallation;
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    this.backgroundExecutor = backgroundExecutor;
+    this.networkExecutor =
+        new ThreadPoolExecutor(
+            /* corePoolSize= */ CORE_POOL_SIZE,
+            /* maximumPoolSize= */ MAXIMUM_POOL_SIZE,
+            /* keepAliveTime= */ 0L,
```

Comment:
did you intend this to be KEEP_ALIVE_TIME_IN_SECONDS?

Reply:
Why is keepAliveTime 0 seconds instead of 30 seconds?